autoload/lsp.vim

```diff
@@ -46,6 +58,67 @@ function! lsp#log(...) abort
     endif
 endfunction
 
+let s:has_lua = has('nvim-0.4.0') || (has('lua') && has('patch-8.2.0775'))
+let s:lua_array_start_index = has('nvim-0.4.0') || has('patch-8.2.1066')
```

Comment:
can we use `lsp#utils#has_lua()` function instead.

Feel free to bump the version if needed as I would like to avoid having start index check except for the diff.

I recent sent a patch to add lambda support so while we are at it we might want to update to minimum vim version to https://github.com/vim/vim/pull/7937 as I'm hoping to use this patch soon.